src/Pap/CampaignHistoryManager.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Pap;
+
+use App\Entity\Pap\BuildingBlock;
+use App\Entity\Pap\BuildingBlockStatistics;
+use App\Entity\Pap\CampaignHistory;
+use App\Entity\Pap\Floor;
+use App\Entity\Pap\FloorStatistics;
+use Doctrine\ORM\EntityManagerInterface;
+
+class CampaignHistoryManager
+{
+    private EntityManagerInterface $em;
+
+    public function __construct(EntityManagerInterface $em)
+    {
+        $this->em = $em;
+    }
+
+    public function createBuildingParts(CampaignHistory $campaignHistory): void
+    {
+        $createdBy = $campaignHistory->getQuestioner();
+        $building = $campaignHistory->getBuilding();
+        if (!$bbName = $campaignHistory->getBuildingBlock()) {
```

Comment:
Change the others refs
```suggestion
        if (!$buildingBlockName = $campaignHistory->getBuildingBlock()) {
```